builder/amazon/chroot/builder.go

```diff
@@ -112,8 +112,8 @@ func (b *Builder) Prepare(raws ...interface{}) ([]string, error) {
 		b.config.MountPath = "/mnt/packer-amazon-chroot-volumes/{{.Device}}"
 	}
 
-	if b.config.MountPartition == 0 {
```

Comment:
my only concern is that if if someone has `0` as their mount partition, the behavior will change. That's probably an invalid config, though, so this type of change should be okay.

The pattern I've liked to see for 3 state configs (0, 1, not set, e.g.) is to use a pointer. If it's nil, you know it's not set and can set it to `1`. if it's `0` you know they want the whole thing